containers/buildah
# ADD: only expand archives at the right time

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

When ADD was used to ADD a directory, the contents of archives that we found inside of it were incorrectly being expanded at the destination for the ADD.

#### How to verify it

Attempt to build https://github.com/intel/sriov-network-device-plugin using its `images/Dockerfile` file and its top-level directory as the build context.  The contents of the `images/ddptool-1.0.0.0.tar.gz` should not be expanded, but if it does, its "makefile" will override the "Makefile" at the top level of the build context, and the build will fail.

Added conformance tests for this.

#### Which issue(s) this PR fixes:

Fixes #2090.
 
#### Special notes for your reviewer:

Going to need this in the 1.16 branch, too.

#### Does this PR introduce a user-facing change?

```
The contents of archives contained in directories being copied using the ADD instruction or `buildah add` should no longer be expanded into the destination directory.
```
